### PR TITLE
add rib filtering

### PR DESCRIPTION
**Pull request by tobikris**
I would like to be able to filter the RIB dump for the originating AS and prefix lengths.

### CODE REVIEW

File: route/route_test.go [hunk 1]
```diff
@@ -666,3 +668,57 @@ func TestRouteEqual(t *testing.T) {
 		assert.Equal(t, tc.equal, res)
 	}
 }
+
+func TestRouteIsBGPOriginatedBy(t *testing.T) {
+	tests := []struct {
+		r      *Route
```
Comment (taktv6) [inline route/route_test.go:674]: Let's add a name/description to each test case.

File: route/route_test.go [hunk 2]
```diff
@@ -666,3 +668,57 @@ func TestRouteEqual(t *testing.T) {
...
+				},
+			},
+			origBy: 65000,
+			isOrig: true,
+		},
+	}
+
+	for _, tc := range tests {
+		res := tc.r.IsBGPOriginatedBy(tc.origBy)
+		assert.Equal(t, tc.isOrig, res)
```
Comment (taktv6) [inline route/route_test.go:722]: Let's print the test cases name here too.

**Comment by tobikris** [comment]
I added test case names and another test for an empty AS path.

**Review by taktv6 — APPROVED** [review]
